File: tests/Search/binary_test.go
package Search

import "testing"
import (
	search "github.com/guldmitry/algorithms/Search"
)

func makeRange(min, max int) []int {
	a := make([]int, max-min+1)
	for i := range a {
		a[i] = min + i
	}
	return a
}

var dataProvider = []struct {
	input            []int
	num              int
	expectedPosition int
}{
	{makeRange(-101, 101), 0, 101},
	{makeRange(-101, 101), -100, 1},
	{makeRange(-101, 101), 1, 102},
	{makeRange(-101, 101), 101, 202},
	{makeRange(-101, 101), 999, -1},
	{[]int{1, 2, 1}, 1, 1},
}

func TestSearch(t *testing.T) {
	for _, v := range dataProvider {
		actualPosition := search.Search(v.input, v.num)
		if v.expectedPosition != actualPosition {
			t.Errorf("Want `%d`, got `%d`", v.expectedPosition, actualPosition)
		}
	}
}

File: tests/Search/BinaryTest.php
<?php

namespace GulDmitry\Algorithms\Tests\Search;

use GulDmitry\Algorithms\Search\Binary;
use PHPUnit\Framework\TestCase;

class BinaryTest extends TestCase
{
    /**
     * @dataProvider dataProvider
     * @param array $searchIn
     * @param int $num
     * @param int|null $expectedPosition
     */
    public function testSearching(array $searchIn, int $num, ?int $expectedPosition)
    {
        $alg = new Binary();

        $actualPosition = $alg->search($searchIn, $num);

        $this->assertEquals($expectedPosition, $actualPosition);
    }

    public function dataProvider()
    {
        $range = range(-101, 101);

        yield [$range, 0, 101];
        yield [$range, -100, 1];
        yield [$range, 1, 102];
        yield [$range, 101, 202];
        yield [$range, 999, null];
        yield [[-22 => 1, 0 => 2, 99 => 3], 2, 1];
        yield [[1, 2, 1], 1, 1];
    }
}
